google-cloud-spanner/src/main/java/com/google/cloud/spanner/SessionClient.java

```diff
@@ -113,30 +114,30 @@ private final class BatchCreateSessionsRunnable implements Runnable {
     private final long channelHint;
     private final int sessionCount;
     private final SessionConsumer consumer;
+    private final Tracer tracer;
 
     private BatchCreateSessionsRunnable(
-        int sessionCount, long channelHint, SessionConsumer consumer) {
+        int sessionCount, long channelHint, SessionConsumer consumer, Tracer tracer) {
       Preconditions.checkNotNull(consumer);
       Preconditions.checkArgument(sessionCount > 0, "sessionCount must be > 0");
       this.channelHint = channelHint;
       this.sessionCount = sessionCount;
       this.consumer = consumer;
+      this.tracer = tracer;
     }
 
     @Override
     public void run() {
       List<SessionImpl> sessions;
       int remainingSessionsToCreate = sessionCount;
-      Span span = SpannerImpl.tracer.spanBuilder(SpannerImpl.BATCH_CREATE_SESSIONS).startSpan();
-      try (Scope s = SpannerImpl.tracer.withSpan(span)) {
-        SpannerImpl.tracer
-            .getCurrentSpan()
-            .addAnnotation(String.format("Creating %d sessions", sessionCount));
+      Span span = tracer.spanBuilder(SpannerImpl.BATCH_CREATE_SESSIONS).startSpan();
+      try (Scope s = tracer.withSpan(span)) {
```

Comment:
OTel's Tracer doesn't have a `withSpan` API - will this change to `span.makeCurrent()` after the switch? (I'm curious if there are other uses of Tracer that will go away in the same way.)

Reply:
Yes I think you're right for any instance where we get/use the current span in context.